### ISSUE TITLE
Incorrect handling of free variables with different variable names across rules

### ISSUE THREAD

**Issue by mttnff** [aprove-developers/LoAT#7]
Failing example:

```
(GOAL COMPLEXITY)
(STARTTERM (FUNCTIONSYMBOLS start))
(VAR a b x y)
(RULES
  start(a,b) -{a,a}> f(x,2*b) [ b > 0 ] 
  start(x,y) -{x,x}> f(b,2*y) [ y < 0 ] 
)
```

The rules are interpreted as:

```
Initial Control flow graph problem:
  Start location: start
      0: start -> f : a'=free, b'=2*b, [ b>0 && a>=0 ], cost: a
      1: start -> f : a'=b, b'=2*b, [ b<0 && a>=0 ], cost: a
```

Where the update `a' = b` is wrong, as `b` is a free (unbound) variable in the second rule.

**Comment by mttnff** [comment]
I guess the problem is that `x` and `y` are renamed to `a` and `b` to unify variable names across rules. But as `b` is already present, it has to be renamed first.

**Comment by mttnff** [comment]
Fixed in aa03536d73c49a3d566698c638970cc041cf9c3c
